From 9602a78fc75f330fc24f24f11aa58ec3b032558e Mon Sep 17 00:00:00 2001
From: Vinu Vaghasia <vinu.vaghasia@amd.com>
Date: Mon, 31 Jan 2022 17:14:05 -0600
Subject: [PATCH 1/1] phosphor-network: Modify active MAC address assign to
 eth0 interface only

active MAC needed to assign eth0 interface for hostname

Signed-off-by: Vinu Vaghasia <vinu.vaghasia@amd.com>
---
 ethernet_interface.cpp | 5 ++++-
 network_manager.cpp    | 5 ++++-
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/ethernet_interface.cpp b/ethernet_interface.cpp
index 55c4073..9928f38 100644
--- a/ethernet_interface.cpp
+++ b/ethernet_interface.cpp
@@ -115,7 +115,10 @@ EthernetInterface::EthernetInterface(sdbusplus::bus::bus& bus,
     // would be same as parent interface.
     if (intfName.find(".") == std::string::npos)
     {
-   activeMACAddress = MacAddressIntf::macAddress(getMACAddress(intfName));
+        if (intfName.compare("eth0") == 0)
+        {
+            activeMACAddress = MacAddressIntf::macAddress(getMACAddress(intfName));
+        }
     }
     EthernetInterfaceIntf::ntpServers(getNTPServersFromConf());

diff --git a/network_manager.cpp b/network_manager.cpp
index b74a59f..9fe4959 100644
--- a/network_manager.cpp
+++ b/network_manager.cpp
@@ -158,7 +158,10 @@ std::string Manager::createInterfaces()
         auto mac = intf->getActiveMACAddress();
         if (mac.compare("00:00:00:00:00:00") != 0)
         {
-            activeMAC = mac;
+            if (interface.compare("eth0") == 0)
+            {
+                activeMAC = mac;
+            }
         }

         this->interfaces.emplace(
--
2.17.1

